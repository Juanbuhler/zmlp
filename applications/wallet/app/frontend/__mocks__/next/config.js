import dotenv from 'dotenv'

dotenv.config()

const getConfig = () => ({
  serverRuntimeConfig: process.env,
  publicRuntimeConfig: process.env,
})

export default getConfig
