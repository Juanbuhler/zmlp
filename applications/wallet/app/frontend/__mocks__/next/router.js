/**
 * on
 */

let mockOnFunction = () => {}

export const __setMockOnFunction = fn => {
  mockOnFunction = fn
}

/**
 * off
 */

let mockOffFunction = () => {}

export const __setMockOffFunction = fn => {
  mockOffFunction = fn
}

/**
 * Router
 */

const Router = {
  events: {
    on: (...args) => {
      mockOnFunction(...args)
    },
    off: (...args) => {
      mockOffFunction(...args)
    },
  },
}

export default Router
