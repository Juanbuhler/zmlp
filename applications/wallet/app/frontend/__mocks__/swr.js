import { createElement } from 'react'

const noop = () => () => {}

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * useSWR
 */

let mockUseSWRResponse = {}

export const __setMockUseSWRResponse = data => {
  mockUseSWRResponse = { revalidate: noop, ...data }
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
