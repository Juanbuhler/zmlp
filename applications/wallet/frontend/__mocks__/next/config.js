const getConfig = () => ({
  serverRuntimeConfig: process.env,
  publicRuntimeConfig: process.env,
})

export default getConfig
