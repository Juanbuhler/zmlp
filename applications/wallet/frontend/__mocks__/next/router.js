/**
 * useRouter
 */

let mockUseRouter = {
  query: {
    action: '',
  },
}

export const __setUseRouter = data => {
  mockUseRouter = data
}

export const useRouter = () => mockUseRouter

/**
 * push
 */

let mockPushFunction = () => {}

export const __setMockPushFunction = fn => {
  mockPushFunction = fn
}

/**
 * on
 */

let mockOnFunction = () => {}

export const __setMockOnFunction = fn => {
  mockOnFunction = fn
}

/**
 * off
 */

let mockOffFunction = () => {}

export const __setMockOffFunction = fn => {
  mockOffFunction = fn
}

/**
 * Router
 */

const Router = {
  push: (...args) => {
    mockPushFunction(...args)
  },
  events: {
    on: (...args) => {
      mockOnFunction(...args)
    },
    off: (...args) => {
      mockOffFunction(...args)
    },
  },
}

export default Router
