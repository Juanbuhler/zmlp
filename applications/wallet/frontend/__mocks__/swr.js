import { createElement } from 'react'

const noop = () => () => {}

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * cache
 */

export const { cache } = jest.requireActual('swr')

/**
 * useSWR
 */

let mockUseSWRResponse = { revalidate: noop, mutate: noop }

export const __setMockUseSWRResponse = (data) => {
  mockUseSWRResponse = { revalidate: noop, mutate: noop, ...data }
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
