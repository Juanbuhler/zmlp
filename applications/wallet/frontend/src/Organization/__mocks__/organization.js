const organization = {
  id: '42869703-fb62-4988-a0d1-e59b15caff06',
  name: 'Alpha',
  plan: 'build',
  projectCount: 3,
  createdDate: '2021-03-18T15:32:43.833370Z',
  modifiedDate: '2021-03-25T16:46:52.160758Z',
}

export default organization
