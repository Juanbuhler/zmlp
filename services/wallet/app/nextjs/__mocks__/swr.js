import { createElement } from 'react'

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * useSWR
 */

let mockUseSWRResponse = {}

export const __setMockUseSWRResponse = data => {
  mockUseSWRResponse = data
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
