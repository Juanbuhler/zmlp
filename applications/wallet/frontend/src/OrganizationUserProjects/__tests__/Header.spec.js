import TestRenderer from 'react-test-renderer'

import organization from '../../Organization/__mocks__/organization'

import OrganizationUserProjectsHeader from '../Header'

const ORGANIZATION_ID = '42869703-fb62-4988-a0d1-e59b15caff06'
const USER_ID = 42

describe('<OrganizationUserProjectsHeader />', () => {
  it('should render properly', () => {
    require('next/router').__setUseRouter({
      pathname: '/organizations/[organizationId]/users/[userId]',
      query: { organizationId: ORGANIZATION_ID, userId: USER_ID },
    })

    require('swr').__setMockUseSWRResponse({ data: organization })

    const component = TestRenderer.create(<OrganizationUserProjectsHeader />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
