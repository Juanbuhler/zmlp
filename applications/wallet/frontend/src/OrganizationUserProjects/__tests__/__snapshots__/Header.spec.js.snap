// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrganizationUserProjectsHeader /> should render properly 1`] = `
Array [
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-top: 24px;
  padding-bottom: 16px;
  padding-right: 4px;
  color: #808080;
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
}

.emotion-3 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-top: 24px;
  padding-bottom: 16px;
  padding-right: 4px;
  color: #808080;
}

.emotion-10 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-top: 24px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    className="emotion-0"
  >
    <div
      className="emotion-1"
    >
      <a
        className="emotion-2"
        href="/organizations"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Organization Admin
      </a>
      <span
        className="emotion-3"
      >
        /
      </span>
    </div>
    <div
      className="emotion-1"
    >
      <a
        className="emotion-2"
        href="/organizations/42869703-fb62-4988-a0d1-e59b15caff06"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Alpha
      </a>
      <span
        className="emotion-3"
      >
        /
      </span>
    </div>
    <div
      className="emotion-1"
    >
      <a
        className="emotion-2"
        href="/organizations/42869703-fb62-4988-a0d1-e59b15caff06/users"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Users
      </a>
      <span
        className="emotion-3"
      >
        /
      </span>
    </div>
    <h2
      className="emotion-10"
    >
      Projects
    </h2>
  </div>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 32px;
  padding: 8px;
  padding-top: 0;
  padding-bottom: 0;
  color: #212121;
  font-size: 11px;
  line-height: 15px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.37px;
}

<div
    className="emotion-0"
  >
    <div
      className="emotion-1"
    >
      build
       plan
    </div>
  </div>,
]
`;
